xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitleStream.h

```diff
@@ -30,12 +32,35 @@ class CDVDSubtitleStream
    */
   bool IsIncompatible(CDVDInputStream* pInputStream, std::vector<uint8_t>& buf, size_t* bytesRead);
 
-  int Read(char* buf, int buf_size);
-  long Seek(long offset, int whence);
+  /*!
+   *  \brief Read some data of specified length, from the current position
+   *  \param length The length of data to be read
+   *  \param data[OUT] The data read
+   *  \return True if success, otherwise false
+   */
+  bool Read(int length, char* data);
```

Comment:
Since all other methods seem to use std::string any reason to rely on c_str() for this particular method?

Reply:
is currenlty used by ValidateSignature in WebVTTHandler.cpp
i need to check how is possible convert that method to use std:string

Reply:
changed with std::string should be good